AppSandboxFileAccess/Classes/AppSandboxFileAccess.m

```diff
@@ -181,15 +187,23 @@ - (BOOL)requestAccessPermissionsForFileURL:(NSURL *)fileURL persistPermission:(B
 	fileURL = [[fileURL URLByStandardizingPath] URLByResolvingSymlinksInPath];
 	
 	// lookup bookmark data for this url, this will automatically load bookmark data for a parent path if we have it
-	NSData *bookmarkData = [AppSandboxFileAccessPersist bookmarkDataForURL:fileURL];
+	NSData *bookmarkData = (self.bookmarkPersistanceDelegate) ? [self.bookmarkPersistanceDelegate bookmarkDataForURL:fileURL] : [AppSandboxFileAccessPersist bookmarkDataForURL:fileURL];
```

Comment:
To reduce the need for the code forks (`if ... else ...` / `... ? ... : ...`) what if we have a default delegate object set, which does the logic for when a custom delegate hasn't been set?